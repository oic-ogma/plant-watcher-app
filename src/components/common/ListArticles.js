import React, { Component } from 'react';
import { ListItem, Error, Spinner } from './';
import { ListView } from 'react-native';

class ListArticles extends Component {
  componentWillMount() {
    this.props.fetchArticles();
    this.createDataSource(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.createDataSource(nextProps);
  }

  createDataSource({ articles }) {
    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2
    });

    this.dataSource = ds.cloneWithRows(articles);
  }

  renderRow({ plantName, articleContents, error }) {
    return <ListItem plantName={plantName} articleContents={articleContents} error={error} />;
  }

  render() {
    const { loading, error } = this.props;

    if (loading) {
      return <Spinner size='large' />;
    }
    if (error) {
      return <Error message={error} />;
    }

    return (
      <ListView
        enableEmptySections
        dataSource={this.dataSource}
        renderRow={this.renderRow}
      />
    );
  }
}

export default ListArticles;
