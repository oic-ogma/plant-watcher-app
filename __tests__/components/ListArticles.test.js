import React from 'react';
import ListArticles from '../../src/components/common/ListArticles';
import renderer from 'react-test-renderer';

describe('記事の表示', () => {
  it('snapshot-正しく画面が呼び出される', () => {
    const INITIAL_STATE = {
      plantName: 'testName',
      loading: false,
      error: '',
      fetchArticles:()=>{},
      articles: [
        {
          articleContents: 'this is test contents. \n this contents is test data.',
          articleId: '-KxRJa7r69I46Xey3MnO',
          plantName: 'テスト',
          uid: 'esyBvhxqwxP5UjaUeQyokRszulJ3',
        },
      ],
    };
    const tree = renderer.create(<ListArticles {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
