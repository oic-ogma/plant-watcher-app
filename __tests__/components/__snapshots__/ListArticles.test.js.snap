// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`記事の表示 snapshot-正しく画面が呼び出される 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 1,
    }
  }
  enableEmptySections={true}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#ddd",
          "borderRadius": 2,
          "borderWidth": 1,
          "elevation": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FF0000",
            "fontSize": 18,
            "paddingTop": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 1,
              "borderColor": "#ddd",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 5,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            テスト
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 1,
              "borderColor": "#ddd",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 5,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            this is test contents. 
             this contents is test data.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
